### PR TITLE
PRL-2494:C7 changes-ther-proceeding

### PR DESCRIPTION
**Pull request by khurshedk**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/app/case/definition.ts [hunk 1]
```diff
@@ -119,9 +119,40 @@ export interface Response {
...
+  value: OtherProceedingDetails;
+}
+
+export interface OtherProceedingDetails {
+  orderDetail: string;
+  caseNo: string;
+  orderDate: Date;
+  currentOrder: YesNoEmpty;
+  orderEndDate: Date;
+  orderCopy: YesNoEmpty;
```
Comment (vivek-sekhar) [inline src/main/app/case/definition.ts:153]: Are these other proceedings not optional details?

File: src/main/steps/tasklistresponse/proceedings/ProceedingDetailsMapper.ts [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+      const orderType = proceedings.value.orderType as ProceedingsOrderTypes;
+      courtProceedingsOrders.push(orderType);
+      req.session.userCase.courtProceedingsOrders = courtProceedingsOrders;
+      let id = 1;
+      proceedings.value.proceedingDetails?.forEach(proceeding => {
+        console.log(proceeding.value.caseNo);
+        console.log(getDisplayDate(proceeding.value.orderDate));
+
+        const proceedingOrderInterface: ProceedingsOrderInterface = {
+          id: getNextId(id),
```
Comment (vivek-sekhar) [inline src/main/steps/tasklistresponse/proceedings/ProceedingDetailsMapper.ts:109]: The name of the function and the actual purpose is not in sync. Probably you can. move increment of the id inside the function itself.

File: src/main/steps/tasklistresponse/proceedings/ProceedingDetailsMapper.ts [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+      req.session.userCase.otherProceedings = {
+        order: proceedingOrderTypeInterface,
+      };
+    });
+  }
+  return req.session.userCase;
+};
+
+function getLocalDate(orderDate: string): Date {
+  if (orderDate['year'] !== '' && orderDate['month'] !== '' && orderDate['day'] !== '') {
```
Comment (vivek-sekhar) [inline src/main/steps/tasklistresponse/proceedings/ProceedingDetailsMapper.ts:134]: Null check can be used instead.

File: src/main/routes.ts [hunk 4]
```diff
@@ -167,6 +171,10 @@ export class Routes {
       );
 
       app.get(`${MIAM_START}/:caseId`, errorHandler(new MIAMGetController(step.view, step.generateContent).get));
+      app.get(
```
Comment (vivek-sekhar) [inline src/main/routes.ts:174]: Why is the route not registered in steps and directly in route files?

File: src/main/steps/tasklistresponse/proceedings/ProceedingPostController.ts [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+      Object.assign(req.session.userCase, caseDataFromCos);
+
+      req.session.userCase?.respondents?.forEach((respondent: Respondent) => {
+        if (respondent?.value?.user?.idamId === req.session?.user.id) {
+          setProceedingDetails(req.session.userCase, respondent, req);
+        }
+      });
+      const caseData = toApiFormat(req?.session?.userCase);
+      caseData.id = caseReference;
+      console.log('*** Case Data **** ' + caseData);
```
Comment (vivek-sekhar) [inline src/main/steps/tasklistresponse/proceedings/ProceedingPostController.ts:36]: Remove unnecessary console statements.

**Review by vivek-sekhar — CHANGES_REQUESTED** [review]

File: src/main/steps/tasklistresponse/proceedings/ProceedingDetailsMapper.ts [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+    respondent.value.response.currentOrPreviousProceedings = currentOrPreviousProceedings;
+  } else {
+    respondent.value.response.currentOrPreviousProceedings = currentOrPreviousProceedings;
+  }
+
+  return respondent;
+};
+
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+export const getProceedingDetails = (respondent: Respondent, req: AppRequest): Partial<CaseWithId> => {
```
Comment (vivek-sekhar) [inline src/main/steps/tasklistresponse/proceedings/ProceedingDetailsMapper.ts:90]: This method is creating side effects.

File: src/main/steps/tasklistresponse/proceedings/ProceedingDetailsMapper.ts [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+  ProceedingsOrderTypeInterface,
+  ProceedingsOrderTypes,
+  Respondent,
+} from '../../../app/case/definition';
+import { AppRequest } from '../../../app/controller/AppRequest';
+
+import { ANYTYPE } from './dateformatter';
+
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+export const setProceedingDetails = (UserCase, respondent: Respondent, req: AppRequest): Respondent => {
```
Comment (vivek-sekhar) [inline src/main/steps/tasklistresponse/proceedings/ProceedingDetailsMapper.ts:20]: This method is creating side effects.

**Review by vivek-sekhar — CHANGES_REQUESTED** [review]

File: src/main/steps/tasklistresponse/proceedings/ProceedingDetailsMapper.ts (hunk 7 above)
Comment (khurshedk) [reply src/main/steps/tasklistresponse/proceedings/ProceedingDetailsMapper.ts:20]: made changes and used local variable as discussed

File: src/main/steps/tasklistresponse/proceedings/ProceedingDetailsMapper.ts (hunk 6 above)
Comment (khurshedk) [reply src/main/steps/tasklistresponse/proceedings/ProceedingDetailsMapper.ts:90]: made changes and used local variable as discussed

File: src/main/steps/tasklistresponse/proceedings/ProceedingPostController.ts (hunk 5 above)
Comment (khurshedk) [reply src/main/steps/tasklistresponse/proceedings/ProceedingPostController.ts]: removed console statement

File: src/main/routes.ts (hunk 4 above)
Comment (khurshedk) [reply src/main/routes.ts:170]: This is followed with similar approach as with other controllers used in.

File: src/main/steps/tasklistresponse/proceedings/ProceedingDetailsMapper.ts (hunk 3 above)
Comment (khurshedk) [reply src/main/steps/tasklistresponse/proceedings/ProceedingDetailsMapper.ts]: added null check as suggested

File: src/main/steps/tasklistresponse/proceedings/ProceedingDetailsMapper.ts (hunk 2 above)
Comment (khurshedk) [reply src/main/steps/tasklistresponse/proceedings/ProceedingDetailsMapper.ts:107]: Made changes as suggested

File: src/main/app/case/definition.ts (hunk 1 above)
Comment (khurshedk) [reply src/main/app/case/definition.ts:153]: Fixed in the latest build

**Review by AlokDatta — APPROVED** [review]

**Pull request merged**